File: CodeBattleCpp/CodeBattleCppLibrary/Point.cpp
#include "Point.h"

Point::Point(int _x, int _y) {
	x = _x;
	y = _y;
}

int Point::getX() {
	return x;
}

int Point::getY() {
	return y;
}


std::string Point::toString() {
	return std::string("[")+ std::to_string(x) +std::string(",")+ std::to_string(y) +std::string("]");
}

File: CodeBattleJava/src/main/java/clientlib/Point.java
package clientlib;

import java.util.Objects;

public class Point {

    protected int x;
    protected int y;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public Point(Point point) {
        this(point.getX(), point.getY());
    }


    public int getX() {
        return x;
    }


    public int getY() {
        return y;
    }

    public void setX(int x) {
        this.x = x;
    }

    public void setY(int y) {
        this.y = y;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Point point = (Point) o;
        return x == point.x &&
                y == point.y;
    }

    @Override
    public int hashCode() {
        return Objects.hash(x, y);
    }
}
